src/Sylius/Bundle/ReviewBundle/EventListener/ReviewChangeListener.php

```diff
@@ -37,13 +37,38 @@ public function __construct(ReviewableRatingUpdaterInterface $averageRatingUpdat
     }
 
     /**
-     * @param GenericEvent $event
+     * @param LifecycleEventArgs $args
      */
-    public function recalculateSubjectRating(GenericEvent $event): void
+    public function postPersist(LifecycleEventArgs $args)
     {
-        $subject = $event->getSubject();
+        $this->recalculateSubjectRating($args);
+    }
+
+    /**
+     * @param LifecycleEventArgs $args
+     */
+    public function postUpdate(LifecycleEventArgs $args)
+    {
+        $this->recalculateSubjectRating($args);
+    }
+
+    /**
+     * @param LifecycleEventArgs $args
+     */
+    public function postRemove(LifecycleEventArgs $args)
+    {
+        $this->recalculateSubjectRating($args);
+    }
+
+    /**
+     * @param LifecycleEventArgs $args
+     */
+    public function recalculateSubjectRating(LifecycleEventArgs $args): void
+    {
+        $subject = $args->getObject();
+
         if (!$subject instanceof ReviewInterface) {
-            throw new UnexpectedTypeException($subject, ReviewInterface::class);
+            return;
         }
 
         if (ReviewInterface::STATUS_ACCEPTED === $subject->getStatus()) {
```

Comment:
What do you think about removing this condition? It won't change the rating if an already accepted review gets declined.

Reply:
@pamil I think it's necessary. Using doctrine lifecycle events, you always have to check you handle wanted entity: http://symfony.com/doc/current/doctrine/event_listeners_subscribers.html

Reply:
What I mean was to leave that `!$subject instanceof ReviewInterface` if, but remove `ReviewInterface::STATUS_ACCEPTED === $subject->getStatus()` one, so when a review is already accepted and you decide to change its status, then the average rating is updated and all logic stays in `AverageRatingUpdater`.

Reply:
yes, you're right. I'm gonna remove that.